relnotes/reverse.feature.md

```diff
@@ -0,0 +1,5 @@
+* `ocean.core.array.Iteration`
+
+  New module containing single function for now - `reverse`, which allows to
+  iterate given slice in reverse order without any allocations (similar to
+  built-in `foreach_reverse`).
```

Comment:
> Most of its downsides is that it doesn't to any type control, happily resorting to plain `foreach` semantics for anything that isn't an array. That makes it rather error-prone in scope of refactorings especially.

That may be true but sounds a bit too speculative to me. I've never seen that happen.
Also, when using `foreach_reverse`, plain `foreach` semantics don't apply to _everything_ that is not an array: `foreach_reverse` over a class or struct that implements `opApply` causes a compile error.